spec/lib/retention_instructions_spec.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+# rubocop:disable RSpec/DescribeClass
+require 'rails_helper'
+require 'rake'
+
+RSpec.describe 'retention_instructions:backfill' do
+
+  shared_examples 'backfilling retention instructions' do
+
+    it 'backfills retention_instruction attribute in labware table' do
+      # Setup
+      labware = create(:labware, retention_instruction: nil,
+                                     custom_metadatum_collection:
+                                       create(:custom_metadatum_collection,
+                                              metadata: { 'retention_instruction' => 'Destroy after 2 years' }
+                                       )
+      )
+
+      # Execute
+      run_rake_task
+
+      # Verify
+      expect(labware.reload.retention_instruction).not_to be_nil
+    end
+
+    it 'removes existing retention instruction from custom_metadata table' do
+      # Setup
+      labware = create(:labware, retention_instruction: nil,
+                                     custom_metadatum_collection:
+                                       create(:custom_metadatum_collection,
+                                              metadata: { 'retention_instruction' => 'Destroy after 2 years' }
+                                       )
+      )
+
+      # Execute
+      run_rake_task
+
+      # Verify
+      expect(labware.reload.custom_metadatum_collection.metadata['retention_instruction']).to be_nil
```

Comment:
Can this be included as part of the test above since they are running the same thing?

Reply:
Hi Ben, so the test above checks for the `labware.retention_instruction` value, while this test checks whether the scripts removes the retention instruction that is currently stored in `custom_metadata`. So, I think the two tests assert for two different things?

Reply:
Hi Ben, so the test above checks for the `labware.retention_instruction` value, while this test checks whether the scripts removes the retention instruction that is currently stored in `custom_metadata`. So, I think the two tests assert for two different things?

Reply:
Yeah thats seems right, even though they assert different things they are outcomes of the same action so could be tested in the same test. That might be personal preference so happy to go with whatever.